### PR TITLE
#4223: Vuetify - Timelines refactoring

### PR DESCRIPTION
**Pull request by ngothikimthaoqb**
- [x] Dense (y/n): add check for dense; add check for body color
- [x] Icon dots: add check for body color
- [x] Small: add check for background title color -> skip - the case has been checked
- [x]  Slots(opposite): add check for opposite text color

### CODE REVIEW

File: jdi-light-vuetify-tests/src/test/java/io/github/epam/vuetify/tests/complex/TimelinesTests.java [hunk 1]
```diff
@@ -72,9 +75,11 @@ public void colorTimeLineDataTest(int index, String time, String event, String c
     @Test
     public void denseTimeLineTest() {
         denseTimeLine.has().size(1);
+        denseTimeLine.is().dense();
         TimeLineItem<IconAlert, UIElement> singleItem = denseTimeLine.item(1);
         singleItem.body().icon().is().type("mdi-information");
         singleItem.body().has().text(LOREM_IPSUM_TEXT);
+        singleItem.body().has().css("color", WHITE.value());
```
Comment (omega790) [inline jdi-light-vuetify-tests/src/test/java/io/github/epam/vuetify/tests/complex/TimelinesTests.java:82]: Please create method "color" to check color

File: jdi-light-vuetify-tests/src/test/java/io/github/epam/vuetify/tests/complex/TimelinesTests.java [hunk 2]
```diff
@@ -72,9 +75,11 @@ public void colorTimeLineDataTest(int index, String time, String event, String c
...
         singleItem.body().has().css("background-color", BLUE.value());
```
Comment (omega790) [inline jdi-light-vuetify-tests/src/test/java/io/github/epam/vuetify/tests/complex/TimelinesTests.java:83]: Please create method "backgroundColor" to check background-color
And add the same for all color checks

File: jdi-light-vuetify-tests/src/test/java/io/github/epam/vuetify/tests/complex/TimelinesTests.java [hunk 3]
```diff
@@ -126,25 +133,30 @@ public void iconTimeLineTest() {
...
-            item.body().has().title("Lorem ipsum");
+            item.body().has().title(LOREM_IPSUM_TITLE);
             item.body().content().has().text(LOREM_IPSUM_TEXT);
         });
     }
 
     @Test(dataProvider = "oppositeTimeLineData", dataProviderClass = TimeLineDataProviders.class)
     public void oppositeTimeLineTest(int index, String date, Colors color) {
         TimeLineItem<DecadeEvent, UIElement> item = oppositeTimeLine.item(index);
+        UIElement title = item.body().find(".headline");
```
Comment (omega790) [inline jdi-light-vuetify-tests/src/test/java/io/github/epam/vuetify/tests/complex/TimelinesTests.java:144]: Please add method to get this part from body

**Comment by ngothikimthaoqb** [comment]
Hi @omega790, I have updated your comments. Could you help me take a look on that? Thank you so much!

File: jdi-light-vuetify-tests/src/test/java/io/github/epam/vuetify/tests/complex/TimelinesTests.java [hunk 4]
```diff
@@ -136,15 +142,19 @@ public void oppositeTimeLineTest(int index, String date, Colors color) {
         TimeLineItem<DecadeEvent, UIElement> item = oppositeTimeLine.item(index);
         item.has().dotColor(color);
         item.opposite().has().text(date);
-        item.opposite().has().css("color", color.value());
+        item.is().oppositeColor( color.value());
```
Comment (omega790) [inline jdi-light-vuetify-tests/src/test/java/io/github/epam/vuetify/tests/complex/TimelinesTests.java:145]: item.is().oppositeColor( color.value()) -> item.opposite().has().color(color.value())
Please create appropriate methods for that check

File: jdi-light-vuetify-tests/src/test/java/io/github/epam/vuetify/tests/complex/TimelinesTests.java [hunk 5]
```diff
@@ -136,15 +142,19 @@ public void oppositeTimeLineTest(int index, String date, Colors color) {
...
+        item.oppositeTitle().has().text(LOREM_IPSUM_TITLE);
```
Comment (omega790) [inline jdi-light-vuetify-tests/src/test/java/io/github/epam/vuetify/tests/complex/TimelinesTests.java:146]: item.oppositeTitle().has().text(LOREM_IPSUM_TITLE) -> item.opposite().title().has().text(LOREM_IPSUM_TITLE);
Please create appropriate methods for that

File: jdi-light-vuetify/src/main/java/com/epam/jdi/light/vuetify/asserts/timelines/TimeLineItemAssert.java [hunk 6]
```diff
@@ -33,4 +33,28 @@ public TimeLineItemAssert dotColor(Enum<?> color) {
         jdiAssert(element().dotColor(), equalTo(color.toString()));
         return this;
     }
+
+    @JDIAction("Assert that color of '{name}' is equal to '{0}'")
+    public TimeLineItemAssert color(String expectedColor) {
+        jdiAssert(element().body().css("color"), Matchers.is(expectedColor));
```
Comment (omega790) [inline jdi-light-vuetify/src/main/java/com/epam/jdi/light/vuetify/asserts/timelines/TimeLineItemAssert.java:39]: The same as a previous

File: jdi-light-vuetify/src/main/java/com/epam/jdi/light/vuetify/asserts/timelines/TimeLineItemAssert.java [hunk 7]
```diff
@@ -33,4 +33,28 @@ public TimeLineItemAssert dotColor(Enum<?> color) {
...
+        return this;
+    }
+
+    @JDIAction("Assert that background color of '{name}' is equal to '{0}'")
+    public TimeLineItemAssert backgroundColor(String expectedColor) {
+        jdiAssert(element().body().css("background-color"), Matchers.is(expectedColor));
```
Comment (omega790) [inline jdi-light-vuetify/src/main/java/com/epam/jdi/light/vuetify/asserts/timelines/TimeLineItemAssert.java:45]: The same as a previous

File: jdi-light-vuetify/src/main/java/com/epam/jdi/light/vuetify/asserts/timelines/TimeLineItemAssert.java [hunk 8]
```diff
@@ -33,4 +33,28 @@ public TimeLineItemAssert dotColor(Enum<?> color) {
...
+        return this;
+    }
+
+    @JDIAction("Assert that opposite title color of '{name}' is equal to '{0}'")
+    public TimeLineItemAssert oppositeTitleColor(String expectedColor) {
+        jdiAssert(element().oppositeTitle().css("color"), Matchers.is(expectedColor));
```
Comment (omega790) [inline jdi-light-vuetify/src/main/java/com/epam/jdi/light/vuetify/asserts/timelines/TimeLineItemAssert.java:51]: The same as a previous

File: jdi-light-vuetify/src/main/java/com/epam/jdi/light/vuetify/asserts/timelines/TimeLineItemAssert.java [hunk 9]
```diff
@@ -33,4 +33,28 @@ public TimeLineItemAssert dotColor(Enum<?> color) {
...
+        return this;
+    }
+
+    @JDIAction("Assert that opposite color of '{name}' is equal to '{0}'")
+    public TimeLineItemAssert oppositeColor(String expectedColor) {
+        jdiAssert(element().opposite().css("color"), Matchers.is(expectedColor));
```
Comment (omega790) [inline jdi-light-vuetify/src/main/java/com/epam/jdi/light/vuetify/asserts/timelines/TimeLineItemAssert.java:57]: The same as a previous

File: jdi-light-vuetify/src/main/java/com/epam/jdi/light/vuetify/asserts/CardAssert.java [hunk 10]
```diff
@@ -29,4 +30,16 @@ public CardAssert subtitle(final Matcher<String> condition) {
     public CardAssert subtitle(String text) {
         return this.subtitle(equalTo(text));
     }
+
+    @JDIAction("Assert that title color of '{name}' is equal to '{0}'")
+    public CardAssert titleColor(String expectedColor) {
+        jdiAssert(element().title().css("color"), Matchers.is(expectedColor));
```
[the remark this answers is not in the document]
Comment (ngothikimthaoqb) [reply jdi-light-vuetify/src/main/java/com/epam/jdi/light/vuetify/asserts/CardAssert.java:36]: Hi @omega790 , "Title" is a element of "Card" class, you mean that I need to create a Custom Card Title class
image

**Comment by ngothikimthaoqb** [comment]
Hi @omega790, Thank you for your feedback, I have updated them already. Please help me take a look!

**Pull request merged**